components/library/Timeline/Preview/Timeline.jsx

```diff
@@ -0,0 +1,91 @@
+import { BlockStack, Box, Icon, InlineGrid, InlineStack, Link, Text } from "@shopify/polaris";
+import { AlertCircleIcon, CheckCircleIcon, ChevronRightIcon, CircleChevronRightIcon } from "@shopify/polaris-icons";
+import styles from './timeline.module.css';
+
+export default function Timeline({ items }) {
+
+  function getBulletIconFromTone(tone) {
+    switch (tone) {
+      case 'critical':
+      case 'caution':
+        return AlertCircleIcon;
+      case 'success':
+        return CheckCircleIcon
+      case 'base':
+      case undefined:
+        return null;
+      default:
+        return CircleChevronRightIcon;
+    }
+  }
+
+  let lastDate = null;
+
+  return (
+    <Box>
+      <BlockStack gap="300">
+        {items?.length ? items.map((item, index) => {
+          const currentDate = item.timestamp.toLocaleDateString([], { year: 'numeric', month: 'long', day: 'numeric' });
+          const showDate = currentDate !== lastDate;
+          lastDate = currentDate;
+          const bulletIcon = getBulletIconFromTone(item.tone);
+
+          return (
+            <>
+              {showDate && (
+                <InlineGrid key={index} gap="200" columns="30px auto 90px">
```

Comment:
You have the key set to the index on both of these `InlineGrid`'s, gives duplicate key console error. You'll want a key on the actual element returned from this .map() though and not on these children, you can just do `<Fragment key={index}` instead of the `<>` shorthand fragments you're doing and remove the `key` props from the `InlineGrid`'s here.